### DISCUSSION TITLE
find jdl or python code sample on how to notify by mail when job ends

### DISCUSSION THREAD

**Discussion by randria** [DIRACGrid/DIRAC#7846] [Q&A] [answered]
Hi all.
The question is in the title. 😄 
I would like to use this class for instance in my script/python job https://github.com/DIRACGrid/DIRAC/blob/integration/src/DIRAC/Core/Utilities/Mail.py
Thanks,

**Comment by fstagni** [comment, marked as answer]
Hi, the DIRAC of doing this is by adding a Worfkflow Module. There is documentation about it in https://dirac.readthedocs.io/en/latest/AdministratorGuide/Systems/Workflow/index.html

In terms of code, I am talking about these modules: https://github.com/DIRACGrid/DIRAC/tree/integration/src/DIRAC/Workflow/Modules
This is a part that is normally extended, for example in LHCb we have an important extension in https://gitlab.cern.ch/lhcb-dirac/LHCbDIRAC/-/tree/master/src/LHCbDIRAC/Workflow/Modules
